File: fact.lua
#!/usr/bin/env lua

-- @@@ get_version: lua -v 2>&1 | awk '{print $2}'

local times = 3000000
local fact_16 = 20922789888000

function fact(n)
    if n == 1 then
        return 1
    end
    return n * fact(n - 1)
end

local ok = true
for index = 1, times, 1 do
    ok = ok and fact(16) == fact_16
end

if ok then
    print("lua finish " .. times .. " - ok")
else
    print("lua finish " .. times .. " - fail")
end

File: fact.php
#!/usr/bin/env php
<?php
    # @@@ get_version: php --version | awk 'FNR == 1 {print $2}'

    $times = 7000000;
    $fact_16 = 20922789888000;

    function fact($n) {
        if ($n < 2) return 1;
        return $n * fact($n - 1);
    }

    $ok = 1;
    for ($i = 0; $i < $times; $i++) {
        $ok = $ok and fact(16) == $fact_16;
    }
    print "php finish $times - " . ($ok ? "ok" : "fail") . "\n";
?>
